src/right-field/RightField.jsx

```diff
@@ -186,15 +206,23 @@ function disableIterator() {
   }
 }
 
+async function fetchQuery(query, eventEmitter) {
+  try {
+    let result = await fetch(`${config.queryFolder}${query.queryLocation}`);
+    return await result.text();
+  } catch (error) {
+    handleQueryFetchFail(error, eventEmitter);
+  }
+}
+
 /**
  * A function that executes a given query and processes every result as a stream based on the EventEmitter.
  * @param {query} query the query which is to be executed
  * @param {EventEmitter} eventEmitter an EventEmitter that listens to and emits UI state changes.
  */
 async function executeQuery(query, eventEmitter) {
   try {
-    let result = await fetch(`${config.queryFolder}${query.queryLocation}`);
-    query.queryText = await result.text();
+    query.queryText = await fetchQuery(query, eventEmitter);
     let fetchFunction = getDefaultSession().info.isLoggedIn ? authFetch : fetch;
```

Comment:
Why is `let` used here?

Reply:
There is no reason for this and I have changed it in [commit 091de4e](https://github.com/SolidLabResearch/generic-data-viewer/pull/24/commits/091de4e880929b4a8c27cfa3a77a75b012576ad4)